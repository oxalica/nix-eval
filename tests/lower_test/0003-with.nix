a: b:
with b.a;
[
  a b foo
  (x: with a; [a x bar])
  (y: with fox; [a x fog])
]
