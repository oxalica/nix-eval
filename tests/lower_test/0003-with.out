Ok(
    Expr(
        Lambda {
            arg: Bind,
            body: Expr(
                Lambda {
                    arg: Bind,
                    body: Expr(
                        LetIn {
                            exprs: [
                                Expr(
                                    Apply {
                                        lambda: Expr(
                                            Apply {
                                                lambda: Expr(
                                                    Builtin(
                                                        GetAttr,
                                                    ),
                                                ),
                                                value: Expr(
                                                    Literal(
                                                        String(
                                                            "a",
                                                        ),
                                                    ),
                                                ),
                                            },
                                        ),
                                        value: Debruijn(
                                            1,
                                        ),
                                    },
                                ),
                            ],
                            body: Expr(
                                List {
                                    items: [
                                        Debruijn(
                                            2,
                                        ),
                                        Debruijn(
                                            1,
                                        ),
                                        Expr(
                                            Apply {
                                                lambda: Expr(
                                                    Apply {
                                                        lambda: Expr(
                                                            Builtin(
                                                                GetAttr,
                                                            ),
                                                        ),
                                                        value: Expr(
                                                            Literal(
                                                                String(
                                                                    "foo",
                                                                ),
                                                            ),
                                                        ),
                                                    },
                                                ),
                                                value: Debruijn(
                                                    0,
                                                ),
                                            },
                                        ),
                                        Expr(
                                            Lambda {
                                                arg: Bind,
                                                body: Expr(
                                                    LetIn {
                                                        exprs: [
                                                            Debruijn(
                                                                4,
                                                            ),
                                                        ],
                                                        body: Expr(
                                                            List {
                                                                items: [
                                                                    Debruijn(
                                                                        4,
                                                                    ),
                                                                    Debruijn(
                                                                        1,
                                                                    ),
                                                                    Expr(
                                                                        Apply {
                                                                            lambda: Expr(
                                                                                Apply {
                                                                                    lambda: Expr(
                                                                                        Builtin(
                                                                                            GetAttr,
                                                                                        ),
                                                                                    ),
                                                                                    value: Expr(
                                                                                        Literal(
                                                                                            String(
                                                                                                "bar",
                                                                                            ),
                                                                                        ),
                                                                                    ),
                                                                                },
                                                                            ),
                                                                            value: Debruijn(
                                                                                0,
                                                                            ),
                                                                        },
                                                                    ),
                                                                ],
                                                            },
                                                        ),
                                                    },
                                                ),
                                            },
                                        ),
                                        Expr(
                                            Lambda {
                                                arg: Bind,
                                                body: Expr(
                                                    LetIn {
                                                        exprs: [
                                                            Expr(
                                                                Apply {
                                                                    lambda: Expr(
                                                                        Apply {
                                                                            lambda: Expr(
                                                                                Builtin(
                                                                                    GetAttr,
                                                                                ),
                                                                            ),
                                                                            value: Expr(
                                                                                Literal(
                                                                                    String(
                                                                                        "fox",
                                                                                    ),
                                                                                ),
                                                                            ),
                                                                        },
                                                                    ),
                                                                    value: Debruijn(
                                                                        2,
                                                                    ),
                                                                },
                                                            ),
                                                        ],
                                                        body: Expr(
                                                            List {
                                                                items: [
                                                                    Debruijn(
                                                                        4,
                                                                    ),
                                                                    Expr(
                                                                        Apply {
                                                                            lambda: Expr(
                                                                                Apply {
                                                                                    lambda: Expr(
                                                                                        Builtin(
                                                                                            GetAttr,
                                                                                        ),
                                                                                    ),
                                                                                    value: Expr(
                                                                                        Literal(
                                                                                            String(
                                                                                                "x",
                                                                                            ),
                                                                                        ),
                                                                                    ),
                                                                                },
                                                                            ),
                                                                            value: Debruijn(
                                                                                0,
                                                                            ),
                                                                        },
                                                                    ),
                                                                    Expr(
                                                                        Apply {
                                                                            lambda: Expr(
                                                                                Apply {
                                                                                    lambda: Expr(
                                                                                        Builtin(
                                                                                            GetAttr,
                                                                                        ),
                                                                                    ),
                                                                                    value: Expr(
                                                                                        Literal(
                                                                                            String(
                                                                                                "fog",
                                                                                            ),
                                                                                        ),
                                                                                    ),
                                                                                },
                                                                            ),
                                                                            value: Debruijn(
                                                                                0,
                                                                            ),
                                                                        },
                                                                    ),
                                                                ],
                                                            },
                                                        ),
                                                    },
                                                ),
                                            },
                                        ),
                                    ],
                                },
                            ),
                        },
                    ),
                },
            ),
        },
    ),
)